### ISSUE TITLE
Breaks with docsify "zoom-image" plugin

### ISSUE THREAD

**Issue by jhildenbiddle** [imyelo/docsify-pagination#1]
I've noticed that this plugin will break docsify's [zoom image](https://docsify.js.org/#/plugins?id=zoom-image) plugin if it is loaded first. This is easy to fix by just switching the `<script>` order, but it shouldn't be necessary. 

**Does not work** (loaded *before* zoom-image.min.js)

```javascript
<script src="https://unpkg.com/docsify-pagination/dist/docsify-pagination.min.js"></script>
<script src="https://unpkg.com/docsify/lib/plugins/zoom-image.min.js"></script>
```

**Works as expected** (loaded *after* zoom-image.min.js)

```javascript
<script src="https://unpkg.com/docsify/lib/plugins/zoom-image.min.js"></script>
<script src="https://unpkg.com/docsify-pagination/dist/docsify-pagination.min.js"></script>
```

Thanks!

**Comment by imyelo** [comment]
I found that if we added an **asynchronous afterEach** hook, the **doneEach** hooks would be broken. —— the **doneEach** event was triggered before the html was rendered to the page. This causes plugins like *zoom-image* to rely on **doneEach** event NOT to work properly. 

As shown in the following docsify source code, **doneEach** events are triggered by the ``next`` function before the page rendering is complete:

https://github.com/QingWei-Li/docsify/blob/e5a263f1bb0909e1629abf76e327f7f2b50340d8/src/core/render/index.js#L149

In addition, I tried to fix this asynchronous process, but found that it was not compatible with the current version of **docsify-pagination**. Our plugin needs to get paging data at **doneEach**, but render it on **afterEach**.

Let me try refactoring this logic into clarity.🤔

**Comment by imyelo** [comment]
The newly released 2.1.0 should fix this issue. I'm closing it. Please feel free to reopen.

**Comment by jhildenbiddle** [comment]
Thank you for the quick turnaround, @imyelo ! I've tested the new version and it's working great.
